### PR TITLE
Change for #128

### PR DESCRIPTION
**Pull request by phillip-kruger**
This is a PR for #128.

- Whole bunch of small changes.
- Broke the asciidoc files up into smaller pieces
- Removed the JsonB section, and moved that into the relevant parts of the spec.
- Removed irrelevant code snippets (like the code for our annotations. That can be seen by looking at the code and has not real value here)
- Changed one test to also test `@Name` on Type, and to make sure a method that only has a `@Source` (so not `@Query`) and has a return type that is not mentioned anywhere else, is still included in the schema.
- Fix (in both spec and TCK) the Locale languge format (en-ZA rather than en_ZA)

### CODE REVIEW

File: spec/src/main/asciidoc/components/mutations.asciidoc [hunk 1]
```diff
@@ -13,31 +13,16 @@
...
-//...
-
-@Target(ElementType.METHOD)
-@Retention(RetentionPolicy.RUNTIME)
-@Documented
-public @interface Mutation {
-    String value() default "";
-}
-----
+The `@Mutation` annotation have the following parameters:
```
[the remark this answers is not in the document]
Comment (phillip-kruger) [reply spec/src/main/asciidoc/components/mutations.asciidoc]: Done

File: spec/src/main/asciidoc/entities/default_values.asciidoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+[[default_values]]
+===== Default Values
```
[the remark this answers is not in the document]
Comment (phillip-kruger) [reply spec/src/main/asciidoc/entities/default_values.asciidoc]: Ok, I linked from the arguments section to this section (as this comes first)

File: spec/src/main/asciidoc/entities/ignorable_fields.asciidoc [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+===== Ignorable fields
+
+There may be cases where a developer wants to use a class as a GraphQL type or input type, but use fields that should
+not be part of the exposed schema. The `@Ignore` annotation can be placed on the field to prevent it from being part of
+the schema.
+
+If the `@Ignore` annotation is placed on the field itself, then the field will be excluded from both the input and 
+output types in the generated schema.  If the annotation is only placed on the "getter" method, then it will only be
+excluded from the input type.  If the annotation is only placed on the "setter" method, then it will only be excluded
+from the output type.
```
[the remark this answers is not in the document]
Comment (phillip-kruger) [reply spec/src/main/asciidoc/entities/ignorable_fields.asciidoc]: Yes !! Good catch. This was here all along !

File: spec/src/main/asciidoc/entities/ignorable_fields.asciidoc [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+Similarly, the `@JsonbTransient` annotation (from Json-B API) can be used to ignore certain fields from the type or input type in the 
+schema. The same rules apply: if the annotation is on the getter, then the field is ignored in the type; if the
+annotation is on the setter, then the field is ignored in the input type; if the annotation is on the Java field, it
```
[the remark this answers is not in the document]
Comment (phillip-kruger) [reply spec/src/main/asciidoc/entities/ignorable_fields.asciidoc]: Yup !

File: spec/src/main/asciidoc/entities/naming.asciidoc [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+[[naming]]
+===== Naming
+
+Users can use the `@Name` annotation (or `@JsonbProperty` from JsonB API) to specify a different field name for the field in the GraphQL
```
[the remark this answers is not in the document]
Comment (phillip-kruger) [reply spec/src/main/asciidoc/entities/naming.asciidoc]: Good catch !

File: spec/src/main/asciidoc/entities/non-nullable_fields.asciidoc [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  #...
+}
+
+input SuperHeroInput {
+  superPowers: [String!]
+  #...
+}
+----
+
+Adding this only on the setter will only mark the input as non null and on the getter the type.
```
[the remark this answers is not in the document]
Comment (phillip-kruger) [reply spec/src/main/asciidoc/entities/non-nullable_fields.asciidoc]: Good catch ! Removed it.

File: spec/src/main/asciidoc/entities/objects.asciidoc [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+Any Plain Old Java Object (POJO) can be an entity.  No special annotations are required. Implementations of MicroProfile
+GraphQL should interpret JSON-B annotations when serializing and deserializing entities to JSON, so it is possible to further define entities
+using JSON-B annotations.
+
+JSON-B annotations can be used to help determine schema information as well as data transformation at runtime. In all
+cases, annotations in the `org.eclipse.microprofile.graphql` package will trump JSON-B annotations if there is a
+conflict.
+
+If the entity cannot be serialized, the implementation must return in an internal server error to the client.
```
[the remark this answers is not in the document]
Comment (phillip-kruger) [reply spec/src/main/asciidoc/entities/objects.asciidoc]: Good point .

File: spec/src/main/asciidoc/entities/objects.asciidoc [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+[source,graphql,numbered]
+----
+type SuperHero {
+    #...
+    currentLocation: String
+    #...
+}
+----
+
+You can also choose to expose the method containing the `@Source` annotation as a top-level `Query` by adding the `@Query` annotation:
```
[the remark this answers is not in the document]
Comment (phillip-kruger) [reply spec/src/main/asciidoc/entities/objects.asciidoc]: I think it use the same example, but this one talk about `@Source` in general and the other about naming a source. I think it's ok like this ??

**Comment by phillip-kruger** [comment]
Hi @andymc12 - all changes done, except the last one. let me know what you think.
Thanks

File: spec/src/main/asciidoc/entities/objects.asciidoc (hunk 8 above)
[the remark this answers is not in the document]
Comment (andymc12) [reply spec/src/main/asciidoc/entities/objects.asciidoc]: yes, that's fine with me.

**Comment by phillip-kruger** [comment]
@andymc12 van we merge?

**Pull request merged**